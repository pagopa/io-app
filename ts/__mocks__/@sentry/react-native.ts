export const captureException = jest.fn();
export const captureMessage = jest.fn();
export const captureEvent = jest.fn();
export const setContext = jest.fn();
export const setExtras = jest.fn();
export const setExtra = jest.fn();
export const setTags = jest.fn();
export const setTag = jest.fn();
export const setUser = jest.fn();
export const lastEventId = jest.fn();
export const captureCheckIn = jest.fn();
export const withMonitor = jest.fn();
export const flush = jest.fn().mockResolvedValue(false);
export const close = jest.fn().mockResolvedValue(false);
export const isInitialized = jest.fn().mockReturnValue(false);
export const isEnabled = jest.fn().mockReturnValue(false);
export const addEventProcessor = jest.fn();
export const startSession = jest.fn();
export const endSession = jest.fn();
export const captureSession = jest.fn();
export const ErrorBoundary = jest.fn(({ children }) => children);
