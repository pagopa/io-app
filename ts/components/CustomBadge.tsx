/**
 * A component to render the circolar badge
 */
import * as React from "react";
import { Circle, Svg } from "react-native-svg";
import customVariables from "../theme/variables";

export class CustomBadge extends React.PureComponent<{}> {
  public render() {
    return (
      <Svg width={14} height={18}>
        <Circle
          cx="50%"
          cy="50%"
          r={14 / 2}
          fill={customVariables.contentPrimaryBackground}
        />
      </Svg>
    );
  }
}
