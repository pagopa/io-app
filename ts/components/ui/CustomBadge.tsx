import { Badge, Text } from "native-base";
import React from "react";
import { StyleProp, ViewStyle } from "react-native";

type Props = {
  badgeStyle: StyleProp<ViewStyle>;
  textStyle: StyleProp<ViewStyle>;
  badgeValue: number;
};

const MAX_BADGE_VALUE = 99;

/**
 * A simple badge used for show the number of messages to read
 *
 */
export default class CustomBadge extends React.PureComponent<Props> {
  public render() {
    return (
      this.props.badgeValue > 0 && (
        <Badge style={this.props.badgeStyle}>
          <Text
            badge={true}
            style={this.props.textStyle}
            accessible={false}
            importantForAccessibility={"no-hide-descendants"}
          >
            {Math.min(this.props.badgeValue, MAX_BADGE_VALUE)}
          </Text>
        </Badge>
      )
    );
  }
}
