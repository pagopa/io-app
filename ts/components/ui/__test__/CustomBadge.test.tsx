import { render } from "@testing-library/react-native";
import React from "react";
import CustomBadge, { customBadgeTestable } from "../CustomBadge";

const testID = "badgeTestID";
describe("CustomBadge component", () => {
  jest.useFakeTimers();
  it(`match snapshot for 0`, () => {
    const component = render(<CustomBadge badgeValue={0} />);
    expect(component).toMatchSnapshot();
  });

  it(`match snapshot for 10`, () => {
    const component = render(<CustomBadge badgeValue={10} />);
    expect(component).toMatchSnapshot();
  });

  it(`match snapshot for 100`, () => {
    const component = render(<CustomBadge badgeValue={100} />);
    expect(component).toMatchSnapshot();
  });

  it(`should be defined for 1 as badge value`, () => {
    const component = render(<CustomBadge badgeValue={1} />).queryByTestId(
      testID
    );
    expect(component).toBeDefined();
  });

  it(`should be not defined for 0 as badge value`, () => {
    const component = render(<CustomBadge badgeValue={0} />).queryByTestId(
      testID
    );
    expect(component).toBeNull();
  });

  it(`should be not defined for undefined as badge value`, () => {
    const component = render(
      <CustomBadge badgeValue={undefined} />
    ).queryByTestId(testID);
    expect(component).toBeNull();
  });

  it(`should have a specific with for 1 digits`, () => {
    const digits = [1, 10, 50, 80, 99, 100, 200, 1000];
    digits.forEach(digit => {
      const component = render(
        <CustomBadge badgeValue={digit} />
      ).queryByTestId(testID);

      expect(component).toBeDefined();
      if (component) {
        expect(component).toHaveStyle({
          width:
            customBadgeTestable!.styles.badgeStyle.width *
            customBadgeTestable!.getWidthMultiplier(digit.toString())
        });
      }
    });
  });
});
