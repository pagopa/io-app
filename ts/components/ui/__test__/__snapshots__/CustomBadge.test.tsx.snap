// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomBadge component match snapshot for 0 1`] = `null`;

exports[`CustomBadge component match snapshot for 10 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#0073E6",
        "borderColor": "#FFFFFF",
        "borderRadius": 10,
        "borderWidth": 2,
        "bottom": 10,
        "height": 20,
        "justifyContent": "center",
        "left": 12,
        "paddingLeft": 0,
        "paddingRight": 0,
        "position": "absolute",
        "width": 20,
      },
      Object {
        "width": 24,
      },
    ]
  }
  testID="badgeTestID"
>
  <Text
    accessible={false}
    importantForAccessibility="no-hide-descendants"
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Titillium Web",
        "fontSize": 10,
        "fontStyle": "normal",
        "fontWeight": "700",
        "textAlign": "center",
      }
    }
  >
    10
  </Text>
</View>
`;

exports[`CustomBadge component match snapshot for 100 1`] = `
<View
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#0073E6",
        "borderColor": "#FFFFFF",
        "borderRadius": 10,
        "borderWidth": 2,
        "bottom": 10,
        "height": 20,
        "justifyContent": "center",
        "left": 12,
        "paddingLeft": 0,
        "paddingRight": 0,
        "position": "absolute",
        "width": 20,
      },
      Object {
        "width": 28,
      },
    ]
  }
  testID="badgeTestID"
>
  <Text
    accessible={false}
    importantForAccessibility="no-hide-descendants"
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "Titillium Web",
        "fontSize": 10,
        "fontStyle": "normal",
        "fontWeight": "700",
        "textAlign": "center",
      }
    }
  >
    99+
  </Text>
</View>
`;
